modules/db/src/main/java/org/opencastproject/db/Activator.java

```diff
@@ -198,6 +191,50 @@ public void start(BundleContext bundleContext) throws Exception {
 
   }
 
+  /**
+   * Attempts to establish a simple connection to the database. This method will retry if the connection attempt fails.
+   *
+   * @param jdbcDriver The JDBC driver class
+   * @param jdbcUrl The JDBC URL
+   * @param jdbcUser The JDBC user
+   * @param jdbcPass The JDBC password
+   */
+  private void waitUntilDatabaseIsAvailable(String jdbcDriver, String jdbcUrl, String jdbcUser, String jdbcPass) {
+    logger.info("Testing connectivity to database at {}", jdbcUrl);
+
+    // Load the JDBC driver for our basic connection test
+    try {
+      Class.forName(jdbcDriver);
+    } catch (ClassNotFoundException e) {
+      logger.error("Couldn't load JDBC Driver {} for initial connection test", jdbcDriver);
+      throw new RuntimeException(e);
+    }
+
+    boolean connectionEstablished = false;
+    try {
+      while (!connectionEstablished) {
+        try (Connection connection = DriverManager.getConnection(jdbcUrl, jdbcUser, jdbcPass)) {
+          connectionEstablished = connection != null && !connection.isClosed();
+        } catch (SQLInvalidAuthorizationSpecException e) {
+          // Invalid credentials
+          String errorMsg = "Couldn't connect to database. Probably invalid database credentials. Check you config.";
+          logger.error(errorMsg);
+          throw new RuntimeException(errorMsg, e);
```

Comment:
It may seem minor, but it's actually pretty annoying to log errors and then throw them. The thrown error will either be caught and handled properly, turning the error into a non-error, or it will stay an error and be logged again, causing the error to appear at two separate places in the logs.

So, just throw the error and do not log:
```suggestion
          throw new RuntimeException("Couldn't connect to database. Probably invalid database credentials. Check you config.";, e);
```

Reply:
I removed it.